CODEOWNERS

```diff
@@ -0,0 +1,4 @@
+* @hashicorp/consul-core-reviewers
```

Comment:
@loshz suggestions on who should be the owner here?

Reply:
LGTM

Reply:
Github complains it's not a valid group for this repo